src/amo/server/base.js

```diff
@@ -346,11 +345,37 @@ function baseServer(
         _log.debug('Second component render after sagas have finished');
 
         const finalHTML = renderHTML({ props, pageProps, store });
-
+        const componentDeclaredStatus = NestedStatus.peek();
```

Comment:
Maybe add a comment to explain that we `peek` to avoid a `rewind`, which would prevent further calls to `rewind`.